clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -143,9 +143,23 @@ public boolean rename(Path path, Path path1) throws IOException {
     }
 
     @Override
-    public boolean delete(Path path, boolean b) throws IOException {
-        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ delete $$$$$$$$$$$$$$$$$$$$$$$$$$$$ ");
-        return false;
+    public boolean delete(Path path, boolean recursive) throws IOException {
+        LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ Delete path {} - recursive {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$",
+                path, recursive);
+
+        ObjectLocation loc = pathToObjectLocation(path);
+        if (loc == null) {
+            throw new FileNotFoundException(path.toString());
```

Comment:
This is the case we wanted `false` in return

Reply:
I'm not sure.  I think this is analogous to S3A receiving an error from AWS when deleting, https://github.com/apache/hadoop/blob/7f93349ee74da5f35276b7535781714501ab2457/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L2746.

Reply:
I removed this null check with my latest commit. Digging into the implementation of pathToObjectLocation I found out that it can't return null values.